### PR TITLE
Adding eval config

### PR DESCRIPTION
**Pull request by thomwolf**
support for lightevals

### CODE REVIEW

File: src/nanotron/config/config.py [hunk 1]
```diff
@@ -375,10 +336,30 @@ def as_dict(self) -> dict:
         return serialize(self)
 
 
-def get_config_from_dict(args: dict, config_class: Type[Config] = Config):
+def get_config_from_dict(
+    config_dict: dict, config_class: Type = Config, skip_unused_config_keys: bool = True, skip_null_keys: bool = True
```
Comment (NouamaneTazi) [inline src/nanotron/config/config.py:340]: maybe we want to default the skips to False?

File: src/nanotron/config/config.py [hunk 2]
```diff
@@ -375,10 +336,30 @@ def as_dict(self) -> dict:
...
+        args: dictionary of arguments
+        config_class: type of the config object to get as a ConfigTypes (Config, LightevalConfig, LightevalSlurm) or str
+        skip_unused_config_keys: whether to skip unused first-level keys in the config file (for config with additional sections)
+        skip_null_keys: whether to skip keys with value None at first and second level
+    """
+    if skip_unused_config_keys:
+        config_dict = {
+            field.name: config_dict[field.name] for field in fields(config_class) if field.name in config_dict
+        }
+    if skip_null_keys:
```
Comment (NouamaneTazi) [inline src/nanotron/config/config.py:354]: should we add warnings when we skip keys or no need to?

**Review by NouamaneTazi — APPROVED** [review]
LGTM modulo minor comments

File: src/nanotron/config/config.py (hunk 2 above)
Comment (thomwolf) [reply src/nanotron/config/config.py:354]: yes sounds good

**Pull request merged**